extensions/sensors/GetEnvironmentalSensors.cpp

```diff
@@ -0,0 +1,156 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <regex.h>
+#include <memory>
+#include <algorithm>
+#include <cctype>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "utils/ByteArrayCallback.h"
+#include "core/FlowFile.h"
+#include "core/logging/Logger.h"
+#include "core/ProcessContext.h"
+#include "core/Relationship.h"
+#include "GetEnvironmentalSensors.h"
+#include "io/DataStream.h"
+#include "io/StreamFactory.h"
+#include "ResourceClaim.h"
+#include "utils/StringUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+std::shared_ptr<utils::IdGenerator> GetEnvironmentalSensors::id_generator_ = utils::IdGenerator::getIdGenerator();
+
+const char *GetEnvironmentalSensors::ProcessorName = "EnvironmentalSensors";
+
+core::Relationship GetEnvironmentalSensors::Success("success", "All files are routed to success");
+
+void GetEnvironmentalSensors::initialize() {
+  logger_->log_trace("Initializing EnvironmentalSensors");
+  // Set the supported properties
+  std::set<core::Property> properties;
+
+  setSupportedProperties(properties);
+  // Set the supported relationships
+  std::set<core::Relationship> relationships;
+  relationships.insert(Success);
+  setSupportedRelationships(relationships);
+}
+
+void GetEnvironmentalSensors::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) {
+
+  imu = RTIMU::createIMU(&settings);
+  if (imu) {
+    imu->IMUInit();
+    imu->setGyroEnable(true);
+    imu->setAccelEnable(true);
+  } else {
+    throw std::runtime_error("RTIMU could not be initialized");
+  }
+
+  humidity_sensor_ = RTHumidity::createHumidity(&settings);
+  if (humidity_sensor_) {
+    humidity_sensor_->humidityInit();
+  } else {
+    throw std::runtime_error("RTHumidity could not be initialized");
+  }
+
+  pressure_sensor_ = RTPressure::createPressure(&settings);
+  if (pressure_sensor_) {
+    pressure_sensor_->pressureInit();
+  } else {
+    throw std::runtime_error("RTPressure could not be initialized");
+  }
+
+}
+
+GetEnvironmentalSensors::~GetEnvironmentalSensors() {
```

Comment:
Possible (but unlikely??) memleak as nothing precludes onSchedule from being called twice. Maybe embed in unique_ptr?